File: backend/app/schemas/signal.py
# backend/app/schemas/signal.py

from pydantic import BaseModel


class SignalItem(BaseModel):
    code: str
    company_name: str | None
    signal_date: str
    target: str
    signal: str  # "BUY" | "HOLD"
    strength: str  # "STRONG" | "WEAK" | "NONE"
    up_prob: float
    pred_rate: float


class SignalListResponse(BaseModel):
    signal_date: str
    target: str
    total: int
    items: list[SignalItem]


class SignalHistoryItem(BaseModel):
    signal_date: str
    signal: str
    strength: str
    up_prob: float
    pred_rate: float


class SignalHistoryResponse(BaseModel):
    code: str
    target: str
    history: list[SignalHistoryItem]

File: frontend/src/types/signal.ts
// frontend/src/types/signal.ts

export type SignalType = "BUY" | "HOLD";
export type StrengthType = "STRONG" | "WEAK" | "NONE" | "ALL";
export type TargetType = "target_1d" | "target_5d" | "target_10d";

export interface SignalItem {
  code: string;
  company_name: string | null;
  signal_date: string;
  target: TargetType;
  signal: SignalType;
  strength: StrengthType;
  up_prob: number;
  pred_rate: number;
}

export interface SignalListResponse {
  signal_date: string;
  target: TargetType;
  total: number;
  items: SignalItem[];
}

export interface SignalHistoryItem {
  signal_date: string;
  signal: SignalType;
  strength: StrengthType;
  up_prob: number;
  pred_rate: number;
}

export interface SignalHistoryResponse {
  code: string;
  target: TargetType;
  history: SignalHistoryItem[];
}